src/containers/StoryList.js

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+
+import { getStoryIds } from "../services/hnAPI";
+
+import { Story } from "../components/Story";
+
+import { StoryListWrapper, ButtonLink } from "../styles/GlobalStyle";
+import { LINKS_PER_PAGE, MAX_PAGES} from "../constants";
+
+
+export default function StoryList({ history }) {
+  const [storyIds, setStoryIds] = useState([]);
+  let { page } = useParams();
+
+  // for pagination
+  let startStory = LINKS_PER_PAGE * +page,
+    endStory = LINKS_PER_PAGE + LINKS_PER_PAGE * +page - 1;
+
+  useEffect(() => {
+    getStoryIds(startStory, endStory).then(
+      data => data && handleStoryIds(data)
+    );
+  }, [page, startStory, endStory]);
+
+  /*
+   * Fixes weird quirk cause by querying the db with orderByKey()
+   * returns a array for for the first call and a object after 2 calls.
+   */
+  function handleStoryIds(ids) {
+    if (typeof ids === "object") {
+      return setStoryIds(Object.values(ids));
+    } else {
+      return setStoryIds(ids);
+    }
+  }
+  // Redirects if the page number is over 17
+  if (+page >= MAX_PAGES) {
+    page = "0";
+    history.push(`/new/0`);
+  }
+
+  function handleNextPage() {
+    if (+page < 17) {
+      history.push(`/new/${+page + 1}`);
+    } else {
+      page = "0";
+      history.push(`/new/0`);
+    }
+  }
```

Comment:
Regarding the page 17 thing:

In my opinion, this is a somewhat dangerous way of handling an edge-case like this, for multiple reasons.

1. It's a bad user experience. If we cannot navigate to page 17, why is there a button to do so?
2. Extending on my comment on the routing logic, the lack of a 404 page is creating confusion. If a manually navigate to `/new/17`, I don't expect to end up on page 0. This is easy to miss too. 
3. This is essentially core routing logic which is hidden inside some component that should not know much about routing. This would be very hard to debug for outside collaborators.

My suggestion to fix this would be:

1. Disable the button if the next page is page 17.
2. Add a 404 page in case someone navigates to it directly.

On top of that, is very confusing to me (from reading the code) why page 17 is special. If the API only returns 500 stories, it would be better to encode this somehow in the check (i.e. `if startStory > 500`) or similar. At least add a comment explaining this case, as it's not quite obvious

Reply:
implemented